### PR TITLE
docs: define java specific waitFor* methods

### PR DESCRIPTION
**Pull request by yury-s**

### CODE REVIEW

File: docs/src/api/java.md [hunk 1]
```diff
@@ -3,6 +3,206 @@
...
+* langs: java
+- returns: <[Frame]>
+
+Runs callback and waits for navigation in one of the frames in the page.
+
+### option: Page.waitForFrameNavigated.timeout = %%-java-wait-for-event-timeout-%%
+
+### param: Page.waitForFrameNavigated.callback = %%-java-wait-for-event-callback-%%
+
+## method: Page.waitForPageError
```
Comment (dgozman) [inline docs/src/api/java.md:86]: Maybe we don't need this one?
Also, I am not sure about waitForFrameAttached, waitForFrameDetached, waitForRequestFailed/Finished, waitForSocketError.

**Review by dgozman — APPROVED** [review]

File: docs/src/api/java.md (hunk 1 above)
Comment (yury-s) [reply docs/src/api/java.md:86]: Well, it is actually used in `TestWorker.shouldReportErrors` test today, without it the test would be more cumbersome. Same with `waitForRequestFailed`. `waitForFrameAttached` does as much sense as the `frameAttached` event itself, so I'd leave it there.

What is your criteria for leaving/removing some of these methods?

Comment (yury-s) [reply docs/src/api/java.md:86]: Ok, I reconsidered this. The error condition should be exceptions rather then something that people put into the expectations and we already have `waitForRequest/Response` methods so adding `waitForRequestFailed` would just confuse people. There are no tests right now that use wait for frame attached/detached, so removing those too. Let's add them if there if there our clients ask for them.

**Comment by yury-s** [comment]
Merged with python's docs, corresponding  playwright.dev change: https://github.com/microsoft/playwright.dev/pull/108

File: docs/src/api/class-browsercontext.md [hunk 2]
```diff
@@ -778,4 +778,21 @@ Event name, same one would pass into `browserContext.on(event)`.
...
+Either a predicate that receives an event or an options object. Optional.
+
+## async method: BrowserContext.waitForPage
+* langs: csharp, java, python
+  - alias-python: expect_page
+- returns: <[Page]>
+
+Performs action and waits for a new [Page] to be created in the context. If predicate is provided, it passes
+[Page] value into the `predicate` function and waits for `predicate(event)` to return a truthy value.
+Will throw an error if the page is closed before the worker event is fired.
```
Comment (dgozman) [inline docs/src/api/class-browsercontext.md:790]: worker -> page

File: docs/src/api/class-page.md [hunk 3]
```diff
@@ -2173,6 +2173,46 @@ Video object associated with this page.
...
+### option: Page.waitForClose.timeout = %%-wait-for-event-timeout-%%
+
+## async method: Page.waitForConsoleMessage
+* langs: csharp, java, python
+  - alias-python: expect_console_message
+- returns: <[ConsoleMessage]>
+
+Performs action and waits for a [ConoleMessage] to be logged by in the page. If predicate is provided, it passes
+[ConsoleMessage] value into the `predicate` function and waits for `predicate(message)` to return a truthy value.
+Will throw an error if the page is closed before the worker event is fired.
```
Comment (dgozman) [inline docs/src/api/class-page.md:2191]: worker -> console

File: docs/src/api/class-page.md [hunk 4]
```diff
@@ -2211,6 +2251,22 @@ frame = event_info.value
...
 Either a predicate that receives an event or an options object. Optional.
 
+## async method: Page.waitForFileChooser
+* langs: csharp, java, python
+  - alias-python: expect_file_chooser
+- returns: <[FileChooser]>
+
+Performs action and waits for a new [FileChooser] to be created. If predicate is provided, it passes
+[FileChooser] value into the `predicate` function and waits for `predicate(fileChooser)` to return a truthy value.
+Will throw an error if the page is closed before the worker event is fired.
```
Comment (dgozman) [inline docs/src/api/class-page.md:2261]: worker -> filechooser

File: docs/src/api/class-page.md [hunk 5]
```diff
@@ -2399,6 +2455,22 @@ Shortcut for main frame's [`method: Frame.waitForNavigation`].
...
+- returns: <[Page]>
+
+Performs action and waits for a popup [Page]. If predicate is provided, it passes
+[Popup] value into the `predicate` function and waits for `predicate(page)` to return a truthy value.
+Will throw an error if the page is closed before the popup event is fired.
+
+### option: Page.waitForPopup.predicate =
+- `predicate` <[function]\([Page]\):[bool]>
+
+Receives the [Popup] object and resolves to truthy value when the waiting should resolve.
```
Comment (dgozman) [inline docs/src/api/class-page.md:2470]: [Popup] -> [Page]

File: docs/src/api/class-websocket.md [hunk 6]
```diff
@@ -54,3 +54,33 @@ Event name, same one would pass into `webSocket.on(event)`.
   - `timeout` <[float]> maximum time to wait for in milliseconds. Defaults to `30000` (30 seconds). Pass `0` to disable timeout. The default value can be changed by using the [`method: BrowserContext.setDefaultTimeout`].
 
 Either a predicate that receives an event or an options object. Optional.
+
+## method: WebSocket.waitForFrameReceived
+* langs: csharp, java
+- returns: <[WebSocketFrame]>
```
Comment (dgozman) [inline docs/src/api/class-websocket.md:60]: What is WebSocketFrame? I see `[string]|[Buffer]` above.

File: docs/src/api/class-websocket.md [hunk 7]
```diff
@@ -54,3 +54,33 @@ Event name, same one would pass into `webSocket.on(event)`.
...
+
+Performs action and waits for a frame to be sent. If predicate is provided, it passes
+[WebSocketFrame] value into the `predicate` function and waits for `predicate(webSocketFrame)` to return a truthy value.
+Will throw an error if the WebSocket or Page is closed before the frame is sent.
```
Comment (dgozman) [inline docs/src/api/class-websocket.md:64]: sent -> received

**Review by dgozman — APPROVED** [review]

File: docs/src/api/class-browsercontext.md (hunk 2 above)
Comment (yury-s) [reply docs/src/api/class-browsercontext.md:790]: Done.

File: docs/src/api/class-page.md (hunk 3 above)
Comment (yury-s) [reply docs/src/api/class-page.md:2191]: Done.

File: docs/src/api/class-websocket.md (hunk 6 above)
Comment (yury-s) [reply docs/src/api/class-websocket.md:60]: There is no way to defined such type union in java so I use a dedicated type for that. Will be happy to change that to something more elegant, let's discuss offline.

**Pull request merged**